app/infrastructure/scrapers/kutis_scraper.py

```
@@ -38,92 +46,90 @@ async def scrape_timetable(
-                        else:
-                            del rowspan_tracker[col_index]
-                        col_index += 1
+        if len(tables) == 3:
+            return tables[1:]
+        return [tables[1]]
+
+    def _parse_timetable(self, table: BeautifulSoup) -> list[dict]:
+        grid = self._build_grid(table)
+        return self._extract_classes_from_grid(grid)
+
+    def _build_grid(self, table: BeautifulSoup):
```

Ruff (0.12.2)

Missing return type annotation for private function `_build_grid`

(ANN202)